weather_stats/weatherstatsapi/weather/openweather/client.py

```diff
@@ -0,0 +1,54 @@
+from types import TracebackType
+from typing import Optional, Type
+
+from aiohttp import ClientSession
+from pydantic import ValidationError
+
+from weather.openweather.exceptions import OpenWeatherError
+from weather.openweather.schemas import GetWeatherData, GetWeatherParams
+from weather.openweather.settings import OPEN_WEATHER_SETTINGS
+
+
+class OpenWeatherClient:
+
+    _session: ClientSession
+
+    async def __aenter__(self) -> "OpenWeatherClient":
```

Comment:
As this project uses python 3.10+, we can simplify `__aenter__` / `__aexit__` by using [@asynccontextmanager](https://docs.python.org/3/library/contextlib.html#contextlib.asynccontextmanager).

Reply:
I don't understand how. Can you give me an example?

Reply:
As discussed offline, this would change API, which is not desired.